Extensions/PlatformBehavior/Extension.cpp

```diff
@@ -758,8 +709,19 @@ void DeclarePlatformBehaviorExtension(gd::PlatformExtension& extension) {
             "string",
             _("Platform type (\"Platform\", \"Jumpthru\" or \"Ladder\")"))
         .MarkAsAdvanced()
-        .SetFunctionName("ChangePlatformType")
-        .SetIncludeFile("PlatformBehavior/PlatformRuntimeBehavior.h");
-#endif
+        .SetFunctionName("ChangePlatformType");
   }
+
+  extension.AddCondition("IsObjectOnGivenFloor",
+                         _("Is object on given floor"),
+                         _("Test if an object is on a given floor."),
+                         _("_PARAM0_ is on floor _PARAM2_"),
+                         _("Platforms"),
+                         "CppPlatform/Extensions/platformicon24.png",
+                         "CppPlatform/Extensions/platformicon16.png")
+           .AddParameter("objectList", _("Object"), "", false)
+           .AddParameter("behavior", _("Behavior"), "PlatformerObjectBehavior")
+           .AddParameter("objectList", _("Platforms"), "", false)
```

Comment:
I have a doubt here @4ian, I don't know if Platforms can be misleading, because this condition actually test if the platformers is on any kind of object.

Reply:
I think it's fair to call them platforms, you can in theory pass any kind of objects, but in this case the condition will simply be false and it will filter out objects that are not the floor, and so that don't even have the platform behavior.

For a user, it makes sense when using this condition to first choose an object, with the platformer behavior, and then to select the "Platform" on which to test as a floor.

Reply:
Why not adding a behavior parameter to constraint the "Platforms" objectList?

Reply:
That's a good question. It turns out that we don't really need it. We could add it to ensure someone select only the objects that can be platform, but it's not even necessary, so we left it without... do you think it would be safer to have it?
Or just to get a validation on the user side?

Reply:
If I am understanding this correctly, we have to decide between these user experiences:

1) User selects an object that doesn't have platform behavior, and wonders why the condition didn't work.

or

2) User tries to select an object that doesn't have platform behavior, and wonders why it is not showing up in the autocompletion in the condition.

Reply:
I think you're correct, it's basically:
1. no check (no behavior parameter), but condition will never be true (because your object is not even a platform, so no chance at all it is a floor)
2. a behavior parameter, so that it will be flagged in red if you select an object that does not have the Platform behavior.

The 2 might actually be better as giving a hint to the user.

Reply:
I agree that #2 is the better user experience.

Reply:
@AlexandreSi If that's ok I think we should make another PR to add this parameter (despite what I initially thought, it's probably better).

Note that in the future, the editor could be smarter and limit the selection to objects having the behavior (right now you can select any object and then BAM the editor is screaming at you with an error telling you that this object does not have the proper behavior).